### PR TITLE
Make sign out a post request via button_to form

### PR DESCRIPTION
**Pull request by steventux**
### Context

Browsers which prefetch pages via links on the current page can unintentionally sign the user out by prefetching the sign out link in the header.

### Changes proposed in this pull request

Extend the header component navigation item to render a sign out form using the `button_to` Rails helper.

image

### Guidance to review

I opted to patching the existing x-govuk header component. We could just as easily introduce our own header component.

### Link to Trello card

https://trello.com/c/XBqfNgbL/1961-make-sign-out-a-post

### Checklist

- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by steventux** [comment]
[Make sign out a POST](https://trello.com/c/XBqfNgbL/1961-make-sign-out-a-post)

File: app/assets/stylesheets/main.scss [hunk 1]
```diff
@@ -5,3 +5,13 @@ $govuk-fonts-path: "/";
 $govuk-images-path: "/";
 
 @import "govuk-frontend/dist/govuk/all";
+
+button.govuk-header__link {
+  background: none !important;
```
Comment (gpeng) [inline app/assets/stylesheets/main.scss:10]: Can these important-ses be avoided by making the selector more specific?

**Review by gpeng — APPROVED** [review]

File: app/assets/stylesheets/main.scss (hunk 1 above)
Comment (steventux) [reply app/assets/stylesheets/main.scss]: Amended to use govuk sass vars and added the correct focus colour

File: lib/govuk_component/header_component.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+class GovukComponent::HeaderComponent
```
Comment (malcolmbaig) [inline lib/govuk_component/header_component.rb:1]: Duplicated class namespacing here

**Pull request merged**